### DISCUSSION TITLE
Problem connecting to the repository on a different port

### DISCUSSION THREAD

**Discussion by socza** [argoproj/argo-cd#10460] [Q&A] [answered]
I added a repository on port other than 22 
git@gitlab.precom.net.pl:\<**different port**\>
```
TYPE  NAME     REPO                                                   MESSAGE  PROJECT
git   test1  git@gitlab.precom.net.pl:**different port**/test1/project-dev1.git  Successful           test1
git   test2  git@gitlab.precom.net.pl:**different port**/test2/project-dev2.git  Successful           test2
```

When I try to add an application in ArgoCD, I get information in the logs that the connection is established on port 22 without respecting the port that is indicated in the added repository.

What is going?
Where can I change the port? Is it even possible?

**Logs**
``level=info msg="finished unary call with code InvalidArgument" error="rpc error: code = InvalidArgument desc = application spec for wakacje-app-test is invalid: InvalidSpecError:  │
│ Unable to generate manifests in install: rpc error: code = Internal desc = Failed to fetch default: `git fetch origin --tags --force` failed exit status 128: ssh: connect to host gitlab.precom.net.pl port 22 │
│ : Connection timed out\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists." grpc.code=InvalidArgument grpc.method=Create grpc.s │
│ ervice=application.ApplicationService grpc.start_time="2022-08-26T09:35:26Z" grpc.time_ms=129601.04 span.kind=server system=grpc ``

**Argo CD Server Version**
Argo CD v2.5.0+37ad433
Build Date 2022-08-25T23:19:48Z
Go Version go1.18.5
Go Compiler gc
Platform linux/amd64
jsonnet v0.18.0
kustomize v4.5.5 2022-05-20T20:25:40Z
Helm v3.9.3+g414ff28
kubect v0.24.2

**Comment by cnkfriday** [comment, marked as answer]
Hi, 
i was using notation like `ssh://git@bitbucket.xxx.pl:7999/project/repo.git` with success. Try this.

**Reply by socza** [reply]
Thx ... works perfect.
